viabel/approximations.py

```diff
@@ -662,8 +674,11 @@ def _pth_moment(self, var_param, p):
         param_dict = self._pattern.fold(var_param)
         D_exp = np.exp(2 * param_dict['log_sigma'])
         B = param_dict['low_rank']
-        Sigma = np.diag(D_exp) + np.dot(B, B.T)
-        Scale = np.linalg.eigvalsh(Sigma)
+
+        #eigvals = eigavalsh(B.T @ B) + eigavalsh(D)
+        eig_bbT = np.linalg.eigvalsh(B @ B.T)
+        eig_d = np.sum(D_exp)
+        Scale = eig_bbT + eig_d
```

Comment:
I don't think you can't just add eigenvalues like this as they aren't in the same basis